### PR TITLE
Refactor Algebras

### LINKED ISSUE #129: Refactor algebras
We should have a criterium about the algebras and where we should place the new ops. Until now, we've created a new algebra per top level menu in [GitHub API](https://developer.github.com/v3/)

In this case, we should:
* Move `Notifications` into a new Algebra called `Activity`
* Move `Statuses` ops into `Repos`

### PR DESCRIPTION
**Pull request by AdrianRaFo**
Fixes #129 Refactor status and notification algebras and their docs.
Create repository documentation

### CODE REVIEW

File: docs/src/main/tut/repo.md [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+layout: docs
+title: Repository API
+---
+
+# Repository API
+
+Github4s supports the [Repository API](https://developer.github.com/v3/repos/). As a result,
+with github4s, you can:
+
+- [Get repository](#get-repository)
```
Comment (BenFradet) [inline docs/src/main/tut/repo.md]: `Get a repository`, no?

File: docs/src/main/tut/repo.md [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+  Some("2014-11-07T22:01:45Z"),
+  Some(Pagination(1, 1000)))
+
+listCommits.exec[cats.Id, HttpResponse[String]]() match {
+  case Left(e) => println("Something went wrong: s{e.getMessage}")
+  case Right(r) => println(r.result)
+}
+```
+
+The `result` on the right is the created [List[Commit]][commit-scala].
```
Comment (BenFradet) [inline docs/src/main/tut/repo.md]: created -> corresponding

File: docs/src/main/tut/repo.md [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+The `result` on the right is a [CombinedStatus][status-scala].
+
+Note that the state of the combined status is the product of a heuristic detailed in
+[the API documentation](https://developer.github.com/v3/repos/statuses/#get-the-combined-status-for-a-specific-ref).
+
+As you can see, a few features of the repository endpoint are missing.
+
+As a result, if you'd like to see a feature supported, feel free to create an issue and/or a pull request!
+
+[repository-scala]: https://github.com/47deg/github4s/blob/master/github4s/`SHA`red/src/main/scala/github4s/free/domain/Repository.scala
```
Comment (BenFradet) [inline docs/src/main/tut/repo.md]: the following links are missing / reference nothing:
- status-scala -> repository-scala
- commit-scala -> repository-scala
- user-scala
- release-scala -> repository-scala
- content-scala -> repository-scala

File: github4s/shared/src/main/scala/github4s/api/Repos.scala [hunk 4]
```diff
@@ -135,7 +135,7 @@ class Repos[C, M[_]](
     )
 
   /**
-   * Fetch contributors list for the the specified repository,
+   * Fetch contributors listStatus for the the specified repository,
```
Comment (BenFradet) [inline github4s/shared/src/main/scala/github4s/api/Repos.scala]: I don't think this is necessary

File: github4s/shared/src/main/scala/github4s/api/Repos.scala [hunk 5]
```diff
@@ -175,7 +175,7 @@ class Repos[C, M[_]](
    * @param targetCommitish specifies the commitish value that determines where the Git tag is created from.
    * Can be any branch or commit SHA. Unused if the Git tag already exists.
    * Default: the repository's default branch (usually `master`).
-   * @param draft `true` to create a draft (unpublished) release, `false` to create a published one.
+   * @param draft `true` to createStatus a draft (unpublished) release, `false` to createStatus a published one.
```
Comment (BenFradet) [inline github4s/shared/src/main/scala/github4s/api/Repos.scala]: same

Comment (BenFradet) [inline github4s/shared/src/main/scala/github4s/api/Repos.scala]: replace all is not a good idea! :smile:

Comment (AdrianRaFo) [inline github4s/shared/src/main/scala/github4s/api/Repos.scala]: it was a problem with refactor rename in intellij, I'm sorry

Comment (BenFradet) [inline github4s/shared/src/main/scala/github4s/api/Repos.scala]: yeah no problem :D

File: github4s/shared/src/main/scala/github4s/free/interpreters/Interpreters.scala [hunk 6]
```diff
@@ -211,15 +225,15 @@ class Interpreters[M[_], C](
   /**
    * Lifts Notification Ops to an effect capturing Monad such as Task via natural transformations
```
Comment (fedefernandez) [inline github4s/shared/src/main/scala/github4s/free/interpreters/Interpreters.scala]: `Notification` -> `Activity`

File: github4s/shared/src/test/scala/github4s/unit/ActivitiesSpec.scala [hunk 7]
```diff
@@ -19,13 +19,13 @@ package github4s.unit
...
-import github4s.api.Notifications
+import github4s.api.Activities
 import github4s.free.domain._
 import github4s.utils.BaseSpec
 
-class NotificationsSpec extends BaseSpec {
+class ActivitiesSpec extends BaseSpec {
 
-  "Notification." should "call to httpClient.put with the right parameters" in {
+  "Activity.SetThreadSubscription" should "call to httpClient.put with the right parameters" in {
```
Comment (fedefernandez) [inline github4s/shared/src/test/scala/github4s/unit/ActivitiesSpec.scala]: `Activity.setThreadSub`

File: github4s/shared/src/test/scala/github4s/unit/GHActivitiesSpec.scala [hunk 8]
```diff
@@ -18,26 +18,26 @@ package github4s.unit
...
+import github4s.api.Activities
 import github4s.app.GitHub4s
 import github4s.free.domain._
 import github4s.utils.BaseSpec
 
-class GHNotificationsSpec extends BaseSpec {
+class GHActivitiesSpec extends BaseSpec {
 
-  "Notifications.SetThreadSubscription" should "call to NotificationOps with the right parameters" in {
+  "Activities.SetThreadSubscription" should "call to ActivityOps with the right parameters" in {
```
Comment (fedefernandez) [inline github4s/shared/src/test/scala/github4s/unit/GHActivitiesSpec.scala]: `Activities.setThreadSub`

File: github4s/shared/src/test/scala/github4s/unit/ReposSpec.scala [hunk 9]
```diff
@@ -87,4 +87,56 @@ class ReposSpec extends BaseSpec {
     )
   }
 
+  "Repos.get" should "call httpClient.get with the right parameters" in {
```
Comment (fedefernandez) [inline github4s/shared/src/test/scala/github4s/unit/ReposSpec.scala]: `Repos.getStatus`

**Pull request merged**